src/Store/auth-context.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+
+const AuthContext = React.createContext({
+    token: '',
+    isLoggedIn: false,
+    login: (token) => { },
+    logout: () => [],
+    isAdmin: false,
+});
+
+export const AuthContextProvider = (props) => {
+    const initialToken = localStorage.getItem('token');
+    const [token, setToken] = useState(initialToken);
+
+    const [admin,setAdmin] = useState(false);
+
+    const userIsLoggedIn = !!token;
+
+    const loginHandler = (props) => {
+        setToken(props.token);
+        localStorage.setItem('token',token);
+
+        if(props.isAdmin==="admin") {
+            setAdmin(true);
+        }
+        else {
+            setAdmin(false);
+        }
+        localStorage.setItem('isAdmin', admin);
+    };
+
+    const logoutHandler = () => {
+        setToken(null);
+        localStorage.removeItem('token');
+    };
+
+
+    const contextValue = {
+        token: token,
+        isLoggedIn: userIsLoggedIn,
+        login: loginHandler,
+        logout: logoutHandler,
+        isAdmin: admin,
+    };
+
+    return (
+        <AuthContext.Provider value={contextValue}>
+            {props.children}
+        </AuthContext.Provider>
+    );
+};
+
+export default AuthContext;
```

Comment:
Even benieuwd waarom we dit in een aparte folder hebben staan (Store) ?

Reply:
Dit is een website breed bestand. Alle andere bestanden zijn components die je heen en weer schuift tussen de lagen. Via useContext kun je iets aanroepen vanaf iedere pagina zonder door alle parents/children te gaan. En voor logIn is dat handig omdat je dat op heel veel pagina's nodig hebt.
En vandaar dus een eigen map, meer gebruik zoals ik het online tegenkom